src/bayesian_linear_regression.jl

```diff
@@ -87,3 +87,17 @@ function __compute_inference_quantities(fx::FiniteBLR, y::AbstractVector{<:Real}
 
     return Uw, Bt, δy, logpdf_δy, Λεy
 end
+
+# Ensure that the posterior weight matrix is a PDMat if the prior is a PDMat.
+__build_Λ(_, U::AbstractMatrix) = Symmetric(U'U)
+__build_Λ(::Type{<:AbstractPDMat}, U::AbstractMatrix) = PDMat(Cholesky(UpperTriangular(U)))
+
+# https://github.com/JuliaLang/julia/pull/39352
+if VERSION < v"1.7"
+    function LinearAlgebra.Cholesky(L::LowerTriangular{T}) where {T}
+        return Cholesky{T,typeof(L.data)}(L.data, 'L', 0)
+    end
+    function LinearAlgebra.Cholesky(U::UpperTriangular{T}) where {T}
+        return Cholesky{T,typeof(U.data)}(U.data, 'U', 0)
+    end
+end
```

Comment:
It's pretty ugly to have to copy this over from ApproxGPs (including tests) - could replace this piracy with an internal function `_chol_from_triu`?

Reply:
This is also causing the codecov failure because coverage is run for 1.7, but it is still being tested on 1.6

Reply:
I agree that this is a bit grim, but I'm happy for it to go in for now -- definitely don't want this package depnding upon ApproximateGPs.